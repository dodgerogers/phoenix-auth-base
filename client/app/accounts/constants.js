import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'GET_CURRENT_USER_REQUEST',
  'GET_CURRENT_USER_SUCCESS',
  'GET_CURRENT_USER_FAILURE',
  'GET_CURRENT_USER_PROFILES_REQUEST',
  'GET_CURRENT_USER_PROFILES_SUCCESS',
  'GET_CURRENT_USER_PROFILES_FAILURE',
  'SET_CURRENT_USER_PROFILE',
  'GET_CURRENT_ACCOUNT_REQUEST',
  'REMOVE_CURRENT_ACCOUNT_REQUEST',
]);
