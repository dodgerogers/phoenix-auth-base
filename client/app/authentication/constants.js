import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'AUTHENTICATE_SUCCESS',
  'AUTHENTICATE_FAILURE',
  'SIGN_OUT_SUCCESS',
  'SIGN_OUT_FAILURE',
  'VERIFY_TOKEN',
  'VERIFY_TOKEN_SUCCESS',
  'VERIFY_TOKEN_FAILURE',
  'PURGE_TOKEN_SUCCESS',
  'PURGE_TOKEN_FAILURE',
  'REGISTER_SUCCESS',
  'REGISTER_FAILURE',
  'CONFIRMATION_SUCCESS',
  'CONFIRMATION_FAILURE',
  'RESEND_CONFIRMATION_SUCCESS',
  'RESEND_CONFIRMATION_FAILURE',
  'PASSWORD_RESET_REQUEST_SUCCESS',
  'PASSWORD_RESET_REQUEST_FAILURE',
]);
