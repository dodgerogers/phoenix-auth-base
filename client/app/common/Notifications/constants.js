import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'NOTIFY',
  'DESTROY',
]);

export const areaIDs = mirrorCreator([
  'APPLICATION',
  'AUTHENTICATION',
]);
