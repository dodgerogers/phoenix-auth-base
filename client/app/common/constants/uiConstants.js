export const colors = {
  RedRibbon: '#E90F3A',
  Carmine: '#a70122',
  RedViolet: '#ea1aa3',
  HaitiPurple: '#120d36',
  ElectricViolet: '#a923e9',
  WindsorPurple: '#580c79',
  ParisMPurple: '#270864',
  PurpleHeart: '#612cc8',
  JacksonsPurple: '#1f28a5',
  CuriousBlue: '#2a8cbd',
  Turquoise: '#2fd6c2',
  Elm: '#1b9487',
  Shamrock: '#2ed583',
  GenoaGreen: '#1a985a',
  Apple: '#2dd535',
  BilbaoGreen: '#138119',
  AtlantisGreen: '#b4d535',
  Olivetone: '#627516',
  HokeyPokeyGold: '#d6aa32',
  ReefGold: '#9e791b',
  Tango: '#ef7d25',
  HawaiianTan: '#a05315',
  Pomegranate: '#ee3b1f',
  Tabasco: '#a72913',
  Sangria: '#8f030a',
  DullLavender: '#9b89e6',
  Crimson: '#e61740',
};
