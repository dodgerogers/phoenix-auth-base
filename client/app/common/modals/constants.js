import mirrorCreator from 'mirror-creator';

export const actionTypes = mirrorCreator([
  'INITIALIZE_MODAL',
  'SHOW_MODAL',
  'HIDE_MODAL',
]);

export const modalIds = mirrorCreator([
  'LOGIN_MODAL',
  'REGISTRATION_MODAL',
  'CONFIRMATION_MODAL',
  'RESEND_CONFIRMATION_MODAL',
  'FORGOT_PASSWORD_MODAL',
  'RESET_PASSWORD_MODAL',
])
