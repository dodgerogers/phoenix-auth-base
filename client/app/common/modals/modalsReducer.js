import { fromJS } from 'immutable';
import { actionTypes } from './modalConstants';

export const initialState = fromJS({
  modals: fromJS({}),
});

export default function modalsReducer(state = initialState, action) {
  const { type } = action;

  switch (type) {
    case actionTypes.SHOW_MODAL:
      return state.set(action.data.id, true);
    case actionTypes.HIDE_MODAL:
      return state.set(action.data.id, false);
    default:
      return state;
  }
}
