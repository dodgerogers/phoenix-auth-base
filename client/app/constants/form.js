import mirrorCreator from 'mirror-creator';

export const formIDs = mirrorCreator([
  'SESSION',
  'REGISTRATION',
  'CONFIRMATION',
  'RESEND_CONFIRMATION',
  'FORGOT_PASSWORD',
  'PASSWORD_RESET',
]);
