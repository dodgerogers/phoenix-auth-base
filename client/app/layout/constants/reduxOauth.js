export const reduxOauthConfig = {
  backend: {
    apiUrl: window.location.href,
    signOutPath:  null, // TODO
    cleanSession: false,
    clientOnly: true,
    tokenValidationPath: 'api/validate_token',
    authProviderPaths: {
      facebook: 'auth/facebook',
      google: 'auth/google_oauth2',
    },
  },
  cookies: document.cookie,
  currentLocation: document.URL
};
