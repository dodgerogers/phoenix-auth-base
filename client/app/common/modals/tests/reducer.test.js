import { fromJS } from 'immutable';
import { areaIDs, actionTypes } from '../constants';
import modalsReducer, { initialState } from '../reducer';


describe('modalsReducer', () => {
  describe('SHOW_MODAL', () => {
    it('toggles modal with ID 1 to true', () => {
      const show = {
        type: actionTypes.SHOW_MODAL,
        data: { id: 1 },
      };

      expect(modalsReducer(initialState, show)).toMatchSnapshot();
    });
  });

  describe('HIDE_MODAL', () => {
    it('toggles modal with ID 1 to false', () => {
      const hide = {
        type: actionTypes.HIDE_MODAL,
        data: { id: 1 },
      };

      expect(modalsReducer(initialState, hide)).toMatchSnapshot();
    });
  });
});
