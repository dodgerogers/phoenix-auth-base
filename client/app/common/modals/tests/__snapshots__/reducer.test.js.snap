// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modalsReducer HIDE_MODAL toggles modal with ID 1 to false 1`] = `
Immutable.Map {
  modals: Immutable.Map {
  },
  1: false,
}
`;

exports[`modalsReducer SHOW_MODAL toggles modal with ID 1 to true 1`] = `
Immutable.Map {
  modals: Immutable.Map {
  },
  1: true,
}
`;
